src/Compilers/VisualBasic/Portable/Symbols/SynthesizedSymbols/SynthesizedParameterSymbol.vb

```diff
@@ -233,6 +233,15 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                 Return _name
             End Get
         End Property
+
+        Friend NotOverridable Overrides Sub AddSynthesizedAttributes(compilationState As ModuleCompilationState, ByRef attributes As ArrayBuilder(Of SynthesizedAttributeData))
+            Dim compilation = Me.DeclaringCompilation
+            If Type.ContainsTupleNames() AndAlso
+                compilation.HasTupleNamesAttributes AndAlso
```

Comment:
:memo: From our discussion, it sounds like we shouldn't get here if the TupleElementNamesAttributes or String types are missing. In such cases, you'd get an error diagnostic (ie prevents you from emitting anyways).
Let's check with Vlad, Aleksey or Andy on what the thinking was.
Either way, we can add those tests (and verify diagnostic).